uberenv.py

```diff
@@ -894,17 +910,42 @@ def disable_spack_config_scopes(self):
                                 "('user', spack.paths.user_config_path)"]:
                 cfg_script = cfg_script.replace(cfg_scope_stmt,
                                                 "#DISABLED BY UBERENV: " + cfg_scope_stmt)
-        open(spack_lib_config,"w").write(cfg_script)
+        # write the updated source
+        open(spack_lib_config_src,"w").write(cfg_script)
+
+        # disable user cache dir
+        spack_cache_path_dir = pjoin(self.dest_dir,"spack_cache")
+        spack_lib_paths_src  = pjoin(self.dest_spack,"lib","spack","spack","paths.py")
+        path_script = open(spack_lib_paths_src).read()
+        print("[disabling user spack cache dir in: {0}]".format(spack_lib_paths_src))
+        spack_ucache_stmt = 'return os.path.expanduser(os.getenv("SPACK_USER_CACHE_PATH") or "~%s.spack" % os.sep)'
+        print(path_script.count(spack_ucache_stmt))
+        if path_script.count(spack_ucache_stmt) > 0:
+            path_script = path_script.replace(spack_ucache_stmt,'return "{0}"'.format(spack_cache_path_dir))
+        # write the updated source
+        open(spack_lib_paths_src,"w").write(path_script)
+
+
+    def set_spack_bootstrap_dir(self):
+        # force bootstrap into dest dir
+        # set bootstrap location in dest dir
+        bstrp_cmd = "{0} bootstrap root --scope=site {1}".format(self.spack_exe(use_spack_env=False),
+                                                                 pjoin(self.dest_dir,"spack_bootstrap"))
+        res = sexe(bstrp_cmd, echo=True)
 
-    def patch(self):
-        # this is an opportunity to show spack python info post obtaining spack
-        self.print_spack_python_info()
 
+    def patch(self):
         # force spack to use only "defaults" config scope
         self.disable_spack_config_scopes()
```

Comment:
we have to disable this first and for most, b/c w/o this -- any command may read/write to`~/.spack`